support setting options for diagonal plots in scatter_matrix

toddrjen opened the pull request:
This adds a new argument that allows the user to specify keyword arguments for the diagonal plots in `hvplot.plotting.scatter_matrix`.  Fixes #599.

jbednar reviewed (approved):
Looks good, thanks! Should we consider also supporting the Pandas `hist_kwds` by code like `if hist_kwds in kwds: diagonal_kwds=kwds.pop('hist_kwds')`? The only reason to do so is if some specific hist_kwds that are valid in Pandas .plot would then be accepted here. If the actual options differ as well, then there'd be no point...

toddrjen commented:
Alright, I added support for both `hist_kwds` and `density_kwds`.  I also added support for specifying a colormap, which is currently hard-coded and cannot be overridden.  The current colormap is okay for `chart='scatter'` but not good for other `chart` types.

toddrjen commented:
The test failures don't seem to be related to this PR. Is there anything else anyone thinks I should change?

philippjfr commented:
Thanks @toddrjen
